express-notes-api/server.js

```diff
@@ -0,0 +1,119 @@
+import express from 'express';
+import { writeFile, readFile } from 'node:fs/promises';
+const app = express();
+const port = 8080;
+
+async function getFiles() {
+  try {
+    const filePath = new URL('data.json', import.meta.url);
+    const content = await readFile(filePath, 'utf8');
+    return content;
+  } catch (error) {
+    console.error('Error');
+  }
+}
+
+async function processFiles(url, content) {
+  try {
+    await writeFile('data.json', content);
+    console.log(content);
+  } catch (error) {
+    console.error('Error');
+  }
+}
+
+var data = JSON.parse(await getFiles('data.json'));
```

Comment:
@rachel-lyy While it may seem more efficient to read the data once at server bootup, that can easily lead to out-of-date data being sent to the client, particularly if there are multiple clients and/or multiple servers accessing the same data source. A better approach is to read a fresh copy of the data with each request so it's always up-to-date.